diff --git a/kong.conf.default b/kong.conf.default
index fa152b7..23dcbf1 100644
--- a/kong.conf.default
+++ b/kong.conf.default
@@ -480,7 +480,7 @@
                                  # To read the file again after modifying it,
                                  # Kong must be reloaded.
 
-#dns_order = LAST,SRV,A,CNAME    # The order in which to resolve different
+#dns_order = LAST,SRV,A,AAAA,CNAME    # The order in which to resolve different
                                  # record types. The `LAST` type means the
                                  # type of the last successful lookup (for the
                                  # specified name). The format is a (case
diff --git a/kong/conf_loader.lua b/kong/conf_loader.lua
index b2bc594..abec02f 100644
--- a/kong/conf_loader.lua
+++ b/kong/conf_loader.lua
@@ -322,7 +322,7 @@ local function check_and_infer(conf)
   end
 
   if conf.dns_order then
-    local allowed = { LAST = true, A = true, CNAME = true, SRV = true }
+    local allowed = { LAST = true, A = true, AAAA = true, CNAME = true, SRV = true }
     for _, name in ipairs(conf.dns_order) do
       if not allowed[name:upper()] then
         errors[#errors+1] = "dns_order: invalid entry '" .. tostring(name) .. "'"
diff --git a/kong/templates/kong_defaults.lua b/kong/templates/kong_defaults.lua
index a556571..73ceb78 100644
--- a/kong/templates/kong_defaults.lua
+++ b/kong/templates/kong_defaults.lua
@@ -65,7 +65,7 @@ db_resurrect_ttl = 30
 
 dns_resolver = NONE
 dns_hostsfile = /etc/hosts
-dns_order = LAST,SRV,A,CNAME
+dns_order = LAST,SRV,A,AAAA,CNAME
 dns_stale_ttl = 4
 dns_not_found_ttl = 30
 dns_error_ttl = 1
